src/Parser/DestructionWarlock/Modules/Features/CooldownTracker.js

```diff
@@ -0,0 +1,56 @@
+import CoreCooldownTracker, { BUILT_IN_SUMMARY_TYPES } from 'Parser/Core/Modules/CooldownTracker';
+
+import SPELLS from 'common/SPELLS';
+
+class CooldownTracker extends CoreCooldownTracker {
+  static cooldownSpells = [
+    ...CooldownTracker.cooldownSpells,
+    {
+      spell: SPELLS.SOUL_HARVEST,
+      summary: [
+        BUILT_IN_SUMMARY_TYPES.DAMAGE,
+      ],
+    },
+    //TODO: see if we can add Havoc (it's not a buff on player, it's a debuff on enemies)
```

Comment:
You can have your own event handlers for activating cooldowns, see the core CooldownTracker's applybuff/removebuff events for info about how to do this. But seeing as it's a debuff it's probably only interesting to see the damage / debuffs done to this target instead of all available enemies, filtering on that might be more challenging (especially to do it with clean code) but can definitely be done.

Reply:
I'll leave it as a future TODO, would be very nice to see how much damage I dealt to the target in that 10 second window, but seeing it would be difficult I'll leave it till later.